crates/proof-of-sql/src/base/database/owned_table.rs

```diff
@@ -132,8 +132,8 @@ impl<S: Scalar> OwnedTable<S> {
     }
     /// Returns the column with the given position.
     #[must_use]
-    pub fn column(&self, index: usize) -> Option<OwnedColumn<S>> {
-        self.table.values().nth(index).cloned()
+    pub fn column_index(&self, index: usize) -> Option<OwnedColumn<S>> {
+        self.table.get_index(index).map(|(_, v)| v).cloned()
     }
```

Comment:
You shouldn't need to clone it if you return `&OwnedColumn`
```suggestion
    pub fn column_index(&self, index: usize) -> Option<&OwnedColumn<S>> {
        self.table.get_index(index).map(|(_, v)| v)
    }
```